### ISSUE TITLE
Prompt extremely slow suddenly

### ISSUE THREAD

**Issue by G2-Games** [ohmybash/oh-my-bash#618]
My Oh My Bash prompt is suddenly taking several hundred ms to display when hitting enter on the terminal. I'm not sure how to debug this. Uninstalling OMB fixes the problem, so it's related in some way.

**Comment by G2-Games** [comment]
I figured the issue out, while spamming the prompt, I saw that the process `hg` was taking up a lot of CPU time, which turns out to be the mercurial command line interface. Uninstalling mercurial (as I don't use it anyway) fixed my problem.

**Comment by RobLoach** [comment]
Which theme were you using? There are a few thmees that implement their own hg prompts.

**Comment by akinomyoga** [comment]
> Which theme were you using? There are a few thmees that implement their own hg prompts.

Most of the themes call `command hg` through `scm_char`, `scm_prompt_info`, etc. in identifying the version control system when `hg` is installed.

https://github.com/ohmybash/oh-my-bash/blob/8cd6496fd09ed8a9a3833679639ba38e3bfd63c6/lib/omb-prompt-base.sh#L120-L129

**Comment by EvansJahja** [comment]
Confirming mercurial is what's causing the issue. As pointed, this line:

`  elif _omb_util_binary_exists hg && [[ -n "$(command hg root 2> /dev/null)" ]]; then SCM=$SCM_HG`

is what's causing slowness, and commenting it makes omb fast.

Not too familiar with the codebase but why don't we make `SCM_CHECK` contains list of scm to check (`"git"`, `"git,svn,hg"`)

**Comment by EvansJahja** [comment]
Actually, why don't wee traverse current and ancestor directory? No cli call required. I only test with git and it works flawlessly #750

**Comment by alexlfyang** [comment]
Agree. I use theme agnoster. It takes around a couple of seconds for prompt.
